components/sup/src/manager/service/mod.rs

```diff
@@ -98,45 +102,32 @@ pub struct Service {
 }
 
 impl Service {
-    fn new(local_member_id: MemberId,
+    fn new(sys: Arc<Sys>,
            package: PackageInstall,
            spec: ServiceSpec,
-           gossip_listen: &GossipListenAddr,
-           http_listen: &http_gateway::ListenAddr,
            manager_fs_cfg: Arc<manager::FsCfg>,
            organization: Option<&str>)
            -> Result<Service> {
         spec.validate(&package)?;
+        let pkg = Pkg::from_install(package)?;
         let spec_file = manager_fs_cfg.specs_path.join(spec.file_name());
-        let service_group = ServiceGroup::new(&package.ident().name, spec.group, organization)?;
-        let runtime_cfg = Self::runtime_config_from(&package)?;
-        let config_root = spec.config_from
-            .clone()
-            .unwrap_or(package.installed_path.clone());
-        let svc_cfg = ServiceConfig::new(&package,
-                                         &runtime_cfg,
-                                         config_root,
-                                         spec.binds.clone(),
-                                         &gossip_listen,
-                                         &http_listen)?;
-        let hook_template_path = svc_cfg.config_root.join("hooks");
-        let hooks_path = fs::svc_hooks_path(service_group.service());
-        let locked_package = Arc::new(RwLock::new(package));
+        let service_group = ServiceGroup::new(&pkg.name, spec.group, organization)?;
+        let config_root = Self::config_root(&pkg, spec.config_from.as_ref().map(|p| &**p));
```

Comment:
This is the best way I've found to convert from an `Option<PathBuf>` to an `Option<&Path>`. The reason is that you need to unwrap the `Option` but there is no way to say unwrap me as `&Path`, only as `&PathBuf`.

With that said, I could just change the function signature that we're calling to accept `Option<&PathBuf`>.  Which would make the calling code a bit cleaner. It's not like this is an API we call very often! I'll make that change

What I *really want to have* is `Option<T> where T: AsRef<Path>` on the function we're calling so we can just pass our `Option<PathBuf>`. I'm sure theres a way to say it in Rust but I just haven't found it yet.